### PR TITLE
8254364: Remove leading _ from struct/union declarations in jvmti.h

### PR DESCRIPTION
**Pull request by iklam**
This PR changes declarations in jvmti.h like the following from

```
struct _jvmtiTimerInfo;
typedef struct _jvmtiTimerInfo jvmtiTimerInfo;
```

to

```
struct jvmtiTimerInfo;
typedef struct jvmtiTimerInfo jvmtiTimerInfo;
```

This way, it becomes possible to make forward declaration in C++ code like this, without assuming the knowledge of the `struct _jvmtiTimerInfo` type:

```
struct jvmtiTimerInfo;
```

Please see bug report [JDK-8254364](https://bugs.openjdk.java.net/browse/JDK-8254364) for the before/after versions of jvmti.h, which is generated from XML.

Tested with build tiers 1-5 in Mach5.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8254364](https://bugs.openjdk.java.net/browse/JDK-8254364): Remove leading _ from struct/union declarations in jvmti.h

### Reviewers
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Serguei Spitsyn](https://openjdk.java.net/census#sspitsyn) (@sspitsyn - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/596/head:pull/596`
`$ git checkout pull/596`

### CODE REVIEW

**Review by dholmes-ora — APPROVED** [review]
Hi Ioi,

A little research indicates the problem of using the same name in the typedef and the structs pre-dates even ANSI C, so this change should be fine.
Please wait for approval from serviceability team member before pushing though.
Thanks,
David

**Review by sspitsyn — APPROVED** [review]
Hi Ioi,

This change looks good to me.

Thanks,
Serguei

**Pull request closed without merging**
